src/Inputs/ComboBox.tsx

```diff
@@ -0,0 +1,241 @@
+    Children,
+    isValidElement,
+} from 'react';
+import styled, { withTheme, DefaultTheme } from 'styled-components';
+import {
+    flex,
+    scroll,
+    position,
+    transition,
+    clickable,
+    darken,
+} from '@Utils/Mixins';
+import { useTransition } from '@Utils/Hooks';
+import { LabelLayout, LabelLayoutProps,InputFragment } from '@Layouts';
+
+const NUM_VISIBLE_SELECTIONS = 3;
+
+var options:any = [];
+
+var isMounted=false;
+
+const SPEED = 'normal';
+
+let myWindow = window as any;
+
+export interface SelectProps extends LabelLayoutProps {
+    disabled?: boolean;
+    placeholder?: string;
+    value?: string | number;
+    theme: DefaultTheme;
+    onChange?: Function;
+    limit?: number;
+    numVisible:number;
+}
+
+const _Select: React.FC<SelectProps> = ({
+    disabled,
+    value,
+    children,
+    limit = 4,
+    placeholder = '',
+    onChange = (): void => {},
+    theme,
+    name,
+    numVisible =NUM_VISIBLE_SELECTIONS,
+    ...props
+}): React.ReactElement => {
+    const [expanded, setExpanded] = useState(false);
+    const [, mount, animation] = useTransition(expanded, {
+        end: theme.speed[SPEED],
+    });
+
+
+
+    const createList = (
+        children: React.ReactNode[],
+        onSelect: React.MouseEventHandler,
+        value?: string | number,
+    ): React.ReactNode[] =>
+        children.map( (child): React.ReactElement | null => {
+
+            if (!myWindow.inputValue){
+                if (child && isValidElement(child)){
+                    const val = child.props.value;
+                    const selected = String(value) === val;
+                    return (
+                        <SelectItem
+                            {...child.props}
+                            active={selected}
+                            onClick={onSelect}
+                            key={val}
+                        />
+                    );}
+            }
+
+            else
+
+            {
+
+            if ((child && isValidElement(child)) && (myWindow.inputValue && (((child.props.children).toLowerCase()).search(myWindow.inputValue.toLowerCase()))> (-1) )) {
+                const val = child.props.value;
+                const selected = String(value) === val;
+                return (
+                    <SelectItem
+                        {...child.props}
+                        active={selected}
+                        onClick={onSelect}
+                        key={val}
+                    />
+                );}
+
+        }
+        
+        return null;} );
+
+    const onSelect = useCallback(
+        ({ currentTarget }): void => {
+            onChange({
+                target: {
+                    value: currentTarget.getAttribute('value'),
+                    name,
+                },
+            });
+            myWindow.inputValue=currentTarget.firstChild.nodeValue;
+        },
+        [name],
+    );
+
+    useEffect((): void | (() => void | undefined) => {

```

Comment:
You do not need to add any extra typings for useEffect, as far as I know

Reply:
unfortunately u do since we have function return being required